### PR TITLE
[Feature] Add the "Undefined team" team shape in a "Supplementary Teams" category

### LINKED ISSUE #6: Add "Undefined Team" as Supplementary Team Type shape
A new Undefined Team Type shape has recently been introduced to enable a consistent representation of teams that have yet to be classified as one of the fundamental team types.

The Miro plugin will need to be updated to include this. See https://github.com/TeamTopologies/Team-Shape-Templates for more details.

### PR DESCRIPTION
**Pull request by malparty**
Close #6 

# What happened?

- [x] Fixed some typos in the README.md file
- [x] Use a newer version of npm
- [x] Add the undefined supplementary team (WIP)

# Proof of work

image

https://user-images.githubusercontent.com/77609814/152129093-78701b60-79f9-4b8f-b19f-06bef03eacf0.mov

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,12 +23,12 @@ Start to drag & drop the shapes you need
 
 ![Start Drag & Drop the needed shapes](readme-img/002.Start_drag_and_drop_new.gif)
 
-Once the needed shapes are in your whiteboard, you can edit them as much as you want (since these are standard Miro shapes):
+Once the needed shapes are into your whiteboard, you can edit them as much as you want (since these are standard Miro shapes):
```
Comment (conjurer-rich) [inline README.md]: "Once you have added the required shapes onto your whiteboard,"

File: src/content-panel.tsx [hunk 2]
```diff
@@ -103,12 +104,18 @@ class Root extends React.Component {
   render() {
     const teamContent = (
       <div className="tt_main_container" onMouseEnter={this.updateCurrentScale}>
-        <h4 className="sub-title">Team types:</h4>
+        <h4 className="sub-title">Fundamental Team types:</h4>
```
Comment (conjurer-rich) [inline src/content-panel.tsx]: Could we arrange the shapes like this?

image

File: src/team-text/team-info.ts [hunk 3]
```diff
@@ -7,6 +7,8 @@ teamText[TEAM_ENUM.Enabling] =
   '**Enabling team:** helps a Stream-aligned team to overcome obstacles. Also detects missing capabilities.'
 teamText[TEAM_ENUM.ComplicatedSubsystem] =
   '**Complicated Subsystem team:** where significant mathematics/calculation/technical expertise is needed.'
+teamText[TEAM_ENUM.UndefinedTeam] =
+  'The **Undefined team type** shape can be used to represent a team that currently has no team type determined.'
```
Comment (conjurer-rich) [inline src/team-text/team-info.ts]: a team that currently has no fundamental team type determined.

**Review by conjurer-rich — COMMENTED** [review]
Looking good, a couple of typos and layout comments

File: src/team-logic/team-supplementary.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          strike: 0,
+          highlighting: ''
+        };
+    }
+  }
+
+  getName(): string {
+    switch (this.teamEnum) {
+      case TEAM_ENUM.UndefinedTeam:
+        return 'Undefined team';
```
Comment (conjurer-rich) [inline src/team-logic/team-supplementary.ts]: "Undefined team type" - since it is the type of the team that is undefined not the team itself 😄

**Comment by conjurer-rich** [comment]
Hey @malparty, when do you think you might be able to make the suggested tweaks? I'd like to write a blog post about the updates so keen to get this merged in.

Thanks

**Comment by malparty** [comment]
> Hey @malparty, when do you think you might be able to make the suggested tweaks? I'd like to write a blog post about the updates so keen to get this merged in.
> 
> Thanks

Hi,
I planned to finish it last weekend but got some unexpected events. I'll try to work on it before the end of the week.

File: README.md (hunk 1 above)
Comment (malparty) [reply README.md]: 8886cb4 ✅

File: src/team-text/team-info.ts (hunk 3 above)
Comment (malparty) [reply src/team-text/team-info.ts]: 9c3854c ✅

File: src/team-logic/team-supplementary.ts (hunk 4 above)
Comment (malparty) [reply src/team-logic/team-supplementary.ts]: 7dbffb4 ✅

File: src/content-panel.tsx (hunk 2 above)
Comment (malparty) [reply src/content-panel.tsx]: UI updated here: c72e216 ✅ 

**Preview**

image

**Comment by malparty** [comment]
@conjurer-rich All updated! ;-) Let me know if OK or need any further change ;-) 👍

**Comment by conjurer-rich** [comment]
Awesome, looking good! Many thanks

**Pull request merged**